src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/DeploymentOptions.cs

```diff
@@ -212,19 +212,23 @@ public class DeploymentOptions
                 ObjectType.AssemblyFiles,
         };
 
-        public DeploymentOptions()
+        public DeploymentOptions(Boolean setDefaults = true)
         {
+
             DacDeployOptions options = new DacDeployOptions();
 
-            // Adding these defaults to ensure behavior similarity with other tools. Dacfx and SSMS import/export wizards use these defaults.
-            // Tracking the full fix : https://github.com/microsoft/azuredatastudio/issues/5599
-            options.AllowDropBlockingAssemblies = true;
-            options.AllowIncompatiblePlatform = true;
-            options.DropObjectsNotInSource = true;
-            options.DropPermissionsNotInSource = true;
-            options.DropRoleMembersNotInSource = true;
-            options.IgnoreKeywordCasing = false;
-            options.IgnoreSemicolonBetweenStatements = false;
+            if (setDefaults)
```

Comment:
I added this for now for tests because a profile without any options will return the default options without setting these. Since we're changing these defaults for deploy, they should also be the defaults for reading from a profile, but I wasn't sure of a good way to do that. One option is to check the profile file to see if these options are set, otherwise set these default values for those that aren't.

Reply:
removed this and decided to set the default values for these options if they aren't specified in the publish profile.